File: resources/数学建模保奖/进阶课2数据预处理/2、数据探索/数据可视化.py

import matplotlib.pyplot as plt

# 假设有两个变量：X和Y
X = [1, 2, 3, 4, 5,15,4,9,1,5,7,3,74,5]
Y = [2, 3, 5, 7, 5,15,4,9,1,5,7,3,74,5]

plt.scatter(X, Y)
plt.title('Scatter Plot Example')
plt.xlabel('X')
plt.ylabel('Y')
plt.show()


import matplotlib.pyplot as plt

# 假设有三个类别的数量
categories = ['A', 'B', 'C']
values = [10, 20, 15]

plt.bar(categories, values)
plt.title('Bar Chart Example')
plt.xlabel('Category')
plt.ylabel('Values')
plt.show()

import matplotlib.pyplot as plt

# 假设有一组数据
data = [1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 9, 10, 10, 10, 11]

plt.boxplot(data)
plt.title('Box Plot Example')
plt.ylabel('Values')
plt.show()

import matplotlib.pyplot as plt

# 假设有一组数据
data = [1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 9, 10, 10, 10, 11]

plt.hist(data, bins=5)
plt.title('Histogram Example')
plt.xlabel('Values')
plt.ylabel('Frequency')
plt.show()

import seaborn as sns
import numpy as np

# 假设有一个相关性矩阵
data = np.random.rand(10, 10)

sns.heatmap(data, annot=True, cmap='coolwarm')
plt.title('Heatmap Example')
plt.show()

import matplotlib.pyplot as plt

# 假设有一组时间序列数据
time = [1, 2, 3, 4, 5]
values = [2, 3, 5, 7, 11]

plt.plot(time, values)
plt.title('Line Plot Example')
plt.xlabel('Time')
plt.ylabel('Values')
plt.show()

File: resources/数学建模保奖/进阶课2数据预处理/2、数据探索/数据可视化.m
% 假设有两个变量：X和Y
X = [1, 2, 3, 4, 5];
Y = [2, 3, 5, 7, 11];

figure;
scatter(X, Y, 'filled');
title('Scatter Plot Example');
xlabel('X');
ylabel('Y');

% 假设有三个类别的数量
categories = {'A', 'B', 'C'};
values = [10, 20, 15];

figure;
bar(values);
set(gca, 'XTickLabel', categories);
title('Bar Chart Example');
xlabel('Category');
ylabel('Values');

% 假设有一组数据
data = [1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 9, 10, 10, 10, 11];

figure;
boxplot(data);
title('Box Plot Example');
ylabel('Values');


% 假设有一组数据
data = [1, 2, 2, 3, 4, 5, 5, 6, 7, 8, 9, 10, 10, 10, 11];

figure;
histogram(data, 5);
title('Histogram Example');
xlabel('Values');
ylabel('Frequency');

% 假设有一个相关性矩阵
data = rand(10, 10);

figure;
imagesc(data);
colorbar;
title('Heatmap Example');

% 假设有一组时间序列数据
time = [1, 2, 3, 4, 5];
values = [2, 3, 5, 7, 11];

figure;
plot(time, values, '-o');
title('Line Plot Example');
xlabel('Time');
ylabel('Values');




